### PR TITLE
4044: Add parent institution relationship to SCO & Program data

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a developer, I need to add a parent relationship to the SCO and Program data so that the database is more normalized.

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/4044

## Testing done
Testing passes locally

## Screenshots
N/A

## Acceptance criteria
- [x] Has_many relationship is established from institution to institution_program.
- [x] Has_many relationship is established from institution to versioned_school_certifying_official.
- [x] Archiver is configured to archive institution_program and versioned_school_certifying_official data.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/institution_builder.rb [hunk 1]
```diff
@@ -89,13 +89,15 @@ def self.initialize_with_weams(version_number)
...
-
+    str = "INSERT INTO institutions (#{columns.join(', ')}, version, created_at, updated_at, version_id) "
+    str += Weam
+           .select(columns)
+           .select("#{version_number.to_i} as version")
+           .select("#{conn.quote(timestamp)} as created_at")
+           .select("#{conn.quote(timestamp)} as updated_at")
+           .select('v.id as version_id')
+           .to_sql
+    str += " JOIN versions v ON v.number = #{version_number}"
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: Please make this INNER JOIN for clarity.

File: app/models/institution_builder.rb [hunk 2]
```diff
@@ -482,16 +485,17 @@ def self.build_zip_code_rates_from_weams(version_number)
       concat_ws(', ', physical_city, physical_state) as physical_location,
       ?,
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: Please make this v.number then pass in version_number once.

File: app/models/institution_builder.rb [hunk 3]
```diff
@@ -482,16 +485,17 @@ def self.build_zip_code_rates_from_weams(version_number)
...
       #{conn.quote(timestamp)},
-      #{conn.quote(timestamp)}
-    FROM weams
+      #{conn.quote(timestamp)},
+      v.id
+      FROM weams JOIN versions v ON v.number = ?
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: INNER JOIN

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Minor changes

**Review by sonntag-adam — APPROVED** [review]

File: db/schema.rb [hunk 4]
```diff
@@ -1612,7 +1617,13 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.integer "version"
+    t.bigint "version_id"
     t.index ["version", "zip_code"], name: "zipcode_rates_archives_version_zip_code_idx"
+    t.index ["version_id"], name: "index_zipcode_rates_archives_on_version_id"
```
Comment (shawkey-daniel) [inline db/schema.rb]: Drop indexes on archives.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by jholton — APPROVED** [review]
LGTM.
Is `VersionedSchoolCertifyingOfficial` going away after this?

**Comment by elder-joseph** [comment]
@jholton In an effort to normalize the data, `VersionedSchoolCertifyingOfficial` could go away after this

**Pull request merged**